src/components/v2/index.ts

```diff
@@ -284,6 +284,13 @@ export default class ComponentV2 extends Component {
         );
       }
       for (const [volumeKey, volumeConfig] of Object.entries(volumes)) {
+        const is_directory = volumeConfig.host_path ? Deno.statSync(volumeConfig.host_path).isDirectory : false;
+        let host_path = undefined;
+        if (volumeConfig.host_path && is_directory) {
+          host_path = path.join(path.dirname(context.component.source), volumeConfig.host_path);
+        } else if (volumeConfig.host_path) {
+          host_path = path.join(context.component.source, volumeConfig.host_path);
+        }
```

Comment:
This feels like you might be intending to check if `context.component.source` is a directory instead of `volumeConfig.host_path`. I assume would be trying to handle the case of the component source being a pointer to an architect.yml file or a folder containing one?